api/auth.go

```diff
@@ -0,0 +1,52 @@
+package api
+
+import (
+	"database/sql"
+	"fmt"
+
+	"github.com/test-network-function/collector/util"
+	"golang.org/x/crypto/bcrypt"
+)
+
+func CheckIfValidCredentials(partnerName, decodePassword string, db *sql.DB) error {
+	// Search for partner in authenticator table
+	var encodedPassword string
+	err := db.QueryRow(util.ExtractPartnerAndPasswordCmd, partnerName).Scan(&encodedPassword)
+	if err != nil {
+		return fmt.Errorf(util.InvalidUsernameErr)
+	}
+
+	// Compare encoded and given passwords
+	err = bcrypt.CompareHashAndPassword([]byte(encodedPassword), []byte(decodePassword))
+	if err != nil {
+		return fmt.Errorf(util.InvalidPasswordErr)
+	}
+
+	return nil
+}
+
+// Already non-empty partner name and decoded password are given
+func VerifyCredentialsAndCreateIfNotExists(partnerName, partnerPassword string, db *sql.DB) error {
+	// Encode the given password
+	encodedPartnerPassword, err := bcrypt.GenerateFromPassword([]byte(partnerPassword), bcrypt.MinCost)
+
+	// Search for partner in authenticator table
+	var encodedPassword string
+	searchPartnerErr := db.QueryRow(util.ExtractPartnerAndPasswordCmd, partnerName).Scan(&encodedPassword)
+
+	// If partner name is not recorded, add partner with encoded password
+	if searchPartnerErr == sql.ErrNoRows {
+		// Create partner entry into the database
+		_, txErr := db.Exec(util.InsertPartnerToAuthSQLCmd, partnerName, encodedPartnerPassword)
+		if txErr != nil {
+			return txErr
+		}
+		return nil
+	}
+	// If partner is found then check if password matches
+	err = bcrypt.CompareHashAndPassword([]byte(encodedPassword), []byte(encodedPartnerPassword))
```

Comment:
Instead of encodedPartnerPassword - has to be partnerPassword.
The methods checks if the "encodedPassword" can be decoded to the "partnerPassword"
We can't just compare the two encoded passwords since the password generated from the two original passwords might defer, even if the origins are equal.

Reply:
So, we decode the encoded password from the db and then compare between partnerPassword and the decoded password, right?

Reply:
Correct. This is the case where the partner's name appear in the authentication table.
We also encode the partnerPassword for the case where the partner name doesn't appear in the table and need to be added. In this case the partner name will be added to the db with the encodedPartnerPassword.

Reply:
Got it, thanks! I updated this part of the code. Please check again.